### PR TITLE
Externalize TTSNotificationProvider and update Spanish Translations

### PR DESCRIPTION
**Pull request by jandrop**

### CODE REVIEW

File: Android/res/values-es/arrays.xml [hunk 1]
```diff
@@ -0,0 +1,270 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (ne0fhyk) [inline Android/res/values-es/arrays.xml]: @jandrop this resource is identical to the one in `values/arrays.xml`, so let's remove it.
Also if strings within the `values/arrays.xml` need to be translated, let's extract them to the `strings.xml` resource, and translate them there.

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java [hunk 2]
```diff
@@ -141,37 +140,36 @@ public void onReceive(Context context, Intent intent) {
...
 
                 case AttributeEvent.MISSION_ITEM_UPDATED:
                     int currentWaypoint = intent.getIntExtra(AttributeEventExtra.EXTRA_MISSION_CURRENT_WAYPOINT, 0);
-					if(currentWaypoint != 0) {
-						//Zeroth waypoint is the home location.
-						speak("Going for waypoint " + currentWaypoint);
-					}
+                    if (currentWaypoint != 0) {
+                        //Zeroth waypoint is the home location.
+                        speak(context.getString(R.string.speak_mission_item_updated) + currentWaypoint);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: @jandrop here you can use [string formatting](https://developer.android.com/reference/java/util/Formatter.html#format(java.lang.String, java.lang.Object...)) to include the current waypoint directly:
- `<string name="speak_mission_item_updated">"Going for waypoint %d"</string>`
- `speak(context.getString(R.string.speak_mission_item_updated, currentWaypoint))`

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java [hunk 3]
```diff
@@ -182,333 +180,340 @@ public void onReceive(Context context, Intent intent) {
...
+                    mMessageBuilder.append(String.format(context.getString(R.string.periodic_status_bat_volt),
+                            droneBattery.getBatteryVoltage()));
+                }
 
-				if (speechPrefs.get(DroidPlannerPrefs.PREF_TTS_PERIODIC_ALT)) {
+                if (speechPrefs.get(DroidPlannerPrefs.PREF_TTS_PERIODIC_ALT)) {
                     final Altitude altitude = drone.getAttribute(AttributeType.ALTITUDE);
-					mMessageBuilder.append("altitude, ").append((int) (altitude.getAltitude())).append(" meters. ");
-				}
+                    mMessageBuilder.append(context.getString(R.string.periodic_status_altitude))
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: @jandrop here also string formatting can be used.

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java [hunk 4]
```diff
@@ -182,333 +180,340 @@ public void onReceive(Context context, Intent intent) {
...
+                            .append(context.getString(R.string.periodic_status_altitude_meters));
+                }
 
-				if (speechPrefs.get(DroidPlannerPrefs.PREF_TTS_PERIODIC_AIRSPEED)) {
+                if (speechPrefs.get(DroidPlannerPrefs.PREF_TTS_PERIODIC_AIRSPEED)) {
                     final Speed droneSpeed = drone.getAttribute(AttributeType.SPEED);
-					mMessageBuilder.append("airspeed, ").append((int) (droneSpeed.getAirSpeed()))
-                            .append(" meters per second. ");
-				}
+                    mMessageBuilder.append(context.getString(R.string.periodic_status_airspeed))
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: @jandrop here also we can use string formatting.

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java [hunk 5]
```diff
@@ -182,333 +180,340 @@ public void onReceive(Context context, Intent intent) {
...
-		if (lastBatteryDischargeNotification > (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT)
-				|| lastBatteryDischargeNotification + 1 < (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT)) {
-			lastBatteryDischargeNotification = (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT);
-			speak("Battery at" + (int) battRemain + "%");
-		}
-	}
-
-	private void speakMode(VehicleMode mode) {
-        if(mode == null)
+                    mMessageBuilder.append(context.getString(R.string.periodic_status_rssi))
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: @jandrop here also we can use string formatting.

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java [hunk 6]
```diff
@@ -182,333 +180,340 @@ public void onReceive(Context context, Intent intent) {
...
+        } else {
+            speak(context.getString(R.string.speak_disarmed));
+        }
+    }
+
+    private void batteryDischargeNotification(double battRemain) {
+        if (lastBatteryDischargeNotification > (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT)
+                || lastBatteryDischargeNotification + 1 < (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT)) {
+            lastBatteryDischargeNotification = (int) ((battRemain - 1) / BATTERY_DISCHARGE_NOTIFICATION_EVERY_PERCENT);
+            speak(context.getString(R.string.speak_battery_notification) + (int) battRemain + "%");
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: @jandrop here also we can use string formatting.

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java [hunk 7]
```diff
@@ -182,333 +180,340 @@ public void onReceive(Context context, Intent intent) {
...
+        }
+    }
+
+    private void speakMode(VehicleMode mode) {
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: @jandrop do these needs to be extracted as well? I'd imagine you'd want the same pronunciation for the flight modes regardless of the language.

File: Android/res/values-es/arrays.xml (hunk 1 above)
Comment (jandrop) [inline Android/res/values-es/arrays.xml]: Ups! in this resource I want only add the lines
 <item>15 seg.</item>
<item>30 seg.</item>
<item>45 seg.</item>
<item>60 seg.</item>

in spanish segundos (seg)

I forgot to delete the rest

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java (hunk 2 above)
Comment (jandrop) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: You're right, my fault. It's been a tough week

File: Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java (hunk 7 above)
Comment (jandrop) [inline Android/src/org/droidplanner/android/notifications/TTSNotificationProvider.java]: I want that when the drone is armed, say "Armado" and when Disarm "Desarmado"

**Comment by ne0fhyk** [comment]
@jandrop thanks for the update. I'll merge and fix the remaining issues.

**Pull request merged**